src/App.tsx

```diff
@@ -1,24 +1,206 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo, FilterType } from './types/Todo';
+import { AddInput } from './components/AddInput/AddInput';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Loading } from './components/isLoading';
+import { TodoList } from './components/TodoList';
+import { AddError } from './components/AddError/AddError';
+import {
+  getTodo,
+  deleteTodo,
+  postTodo,
+  updateTodo,
+} from './api/todos';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 7025;
 
 export const App: React.FC = () => {
+  const [todos, setTodo] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisableInput, setIsDisableInput] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const completedTodos = todos.every((todo) => todo.completed);
+
+  const fetchData = async () => {
+    const todosFromServer = await getTodo(USER_ID);
+
+    try {
+      setTodo(todosFromServer);
+      setIsLoading(true);
+    } catch {
+      setError('Unable to load a todo');
+      setIsLoading(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const filterTodo = useMemo(() => {
+    return todos.filter((todo) => {
+      switch (filterType) {
+        case FilterType.ALL:
+          return todo;
+
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filterType]);
+
+  const handleAddTodo = async (title: string) => {
+    try {
+      setIsDisableInput(true);
+
+      const data = {
+        id: Math.floor(Math.random() * 100),
```

Comment:
My guess it's tha main reason for error while adding todo use crypto.randomUUID()
```suggestion
        id: crypto.randomUUID(),
```